packages/ui/src/utils/__tests__/timePeriodUtils.test.ts

```diff
@@ -0,0 +1,322 @@
+        expect(PERIOD_TYPES.YEAR).toBe('YEAR');
+        expect(PERIOD_TYPES.ANY).toBe('ANY');
+    });
+});
+
+describe('toDHIS2PeriodData', () => {
+    describe('month periods', () => {
+        it('should generate correct month periods for a single month', () => {
+            const result = toDHIS2PeriodData('2024-01', '2024-01', PERIOD_TYPES.MONTH);
+            expect(result).toHaveLength(1);
+            expect(result[0].id).toBe('202401');
+        });
+
+        it('should generate correct month periods for multiple months', () => {
+            const result = toDHIS2PeriodData('2024-01', '2024-03', PERIOD_TYPES.MONTH);
+            expect(result).toHaveLength(3);
+            expect(result[0].id).toBe('202401');
+            expect(result[1].id).toBe('202402');
+            expect(result[2].id).toBe('202403');
+        });
+
+        it('should generate correct month periods spanning years', () => {
+            const result = toDHIS2PeriodData('2023-11', '2024-02', PERIOD_TYPES.MONTH);
+            expect(result).toHaveLength(4);
+            expect(result[0].id).toBe('202311');
+            expect(result[1].id).toBe('202312');
+            expect(result[2].id).toBe('202401');
+            expect(result[3].id).toBe('202402');
+        });
+
+        it('should include correct start and end dates for each month period', () => {
+            const result = toDHIS2PeriodData('2024-01', '2024-01', PERIOD_TYPES.MONTH);
+            expect(result[0].startDate).toEqual(new Date('2024-01-01'));
+            expect(result[0].endDate).toEqual(new Date('2024-01-31'));
+        });
+
+        it('should have correct start and end dates for February in a leap year', () => {
+            const result = toDHIS2PeriodData('2024-02', '2024-02', PERIOD_TYPES.MONTH);
+            expect(result[0].startDate).toEqual(new Date('2024-02-01'));
+            expect(result[0].endDate).toEqual(new Date('2024-02-29'));
+        });
+    });
+
+    describe('week periods', () => {
+        it('should generate correct week periods for a single week', () => {
+            const result = toDHIS2PeriodData('2024-W01', '2024-W01', PERIOD_TYPES.WEEK);
+            expect(result).toHaveLength(1);
+            expect(result[0].id).toBe('2024W1');
+        });
+
+        it('should generate correct week periods for multiple weeks', () => {
+            const result = toDHIS2PeriodData('2024-W01', '2024-W03', PERIOD_TYPES.WEEK);
+            expect(result).toHaveLength(3);
+            expect(result[0].id).toBe('2024W1');
+            expect(result[1].id).toBe('2024W2');
+            expect(result[2].id).toBe('2024W3');
+        });
+
+        it('should include correct start and end dates for each week period', () => {
+            // ISO week 1 of 2024 starts on Monday 2024-01-01 and ends on Sunday 2024-01-07
+            const result = toDHIS2PeriodData('2024-W01', '2024-W01', PERIOD_TYPES.WEEK);
+            expect(result[0].startDate).toEqual(new Date('2024-01-01'));
```

Comment:
can you move these assertions to the test on line 60, and instead make this test check for start and end dates that cross a year (e.g. 2023, 2024)